### PR TITLE
Only patch SVC_Call in "real" code, not in rodata section

### PR DESCRIPTION
**Pull request by cbouvet-ledger**

### CODE REVIEW

File: src/launcher.c [hunk 1]
```diff
@@ -406,6 +446,11 @@ static int load_cxlib(char *cxlib_args)
     }
   }
 
+  if (mprotect(p, sh_size, PROT_READ | PROT_WRITE) != 0) {
+    warn("mprotect(PROT_READ | PROT_WRITE)");
```
Comment (fbeutin-ledger) [inline src/launcher.c:450]: Maybe be more explicit in the warning or add a comment about what this code does

File: src/launcher.c [hunk 2]
```diff
@@ -414,6 +459,11 @@ static int load_cxlib(char *cxlib_args)
     return -1;
   }
 
+  if (mprotect(p, sh_size, PROT_READ | PROT_EXEC) != 0) {
+    warn("mprotect(PROT_READ | PROT_EXEC)");
```
Comment (fbeutin-ledger) [inline src/launcher.c:463]: Maybe be more explicit in the warning or add a comment about what this code does

File: src/launcher.c [hunk 3]
```diff
@@ -293,10 +314,29 @@ static void *load_app(char *name)
...
+    warn("mprotect(PROT_READ | PROT_WRITE)");
     goto error;
   }
 
+  if (patch_svc(code + start, code_length) != 0) {
+    goto error;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_EXEC) != 0) {
+    warn("mprotect(PROT_READ | PROT_EXEC)");
```
Comment (fbeutin-ledger) [inline src/launcher.c:336]: Maybe be more explicit in the warning or add a comment about what this code does

File: src/launcher.c [hunk 4]
```diff
@@ -293,10 +314,29 @@ static void *load_app(char *name)
...
+
+  uint32_t start = 0;
+  uint32_t code_length = size;
+  if (app->elf.code_start && app->elf.code_end) {
+    start = app->elf.code_start - 0xc0de0000;
+    code_length = app->elf.code_end - app->elf.code_start;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_WRITE) != 0) {
+    warn("mprotect(PROT_READ | PROT_WRITE)");
```
Comment (fbeutin-ledger) [inline src/launcher.c:327]: Maybe be more explicit in the warning or add a comment about what this code does

File: src/launcher.c [hunk 5]
```diff
@@ -293,10 +314,29 @@ static void *load_app(char *name)
...
     goto error;
   }
 
+  if (patch_svc(code + start, code_length) != 0) {
+    goto error;
```
Comment (fbeutin-ledger) [inline src/launcher.c:332]: Add a warning

File: src/launcher.c [hunk 6]
```diff
@@ -293,10 +314,29 @@ static void *load_app(char *name)
     }
   }
 
-  if (patch_svc(code, size) != 0) {
+
+
+  uint32_t start = 0;
+  uint32_t code_length = size;
+  if (app->elf.code_start && app->elf.code_end) {
+    start = app->elf.code_start - 0xc0de0000;
```
Comment (fbeutin-ledger) [inline src/launcher.c:322]: Maybe use a define for `0xc0de0000`

**Review by xchapron-ledger — DISMISSED** [review]

File: src/launcher.c [hunk 7]
```diff
@@ -293,10 +314,29 @@ static void *load_app(char *name)
     }
   }
 
-  if (patch_svc(code, size) != 0) {
+
+
+  uint32_t start = 0;
+  uint32_t code_length = size;
+  if (app->elf.code_start && app->elf.code_end) {
```
Comment (fbeutin-ledger) [inline src/launcher.c:321]: `app->elf.code_start != 0 && app->elf.code_end != 0`
I think that implicit boolean cast for integers is confusing, this is more explicit and clear

**Comment by xchapron-ledger** [comment]
You just have some lint issue :)

File: src/launcher.c [hunk 8]
```diff
@@ -166,12 +170,29 @@ int replace_current_code(struct app_s *app)
     return -1;
   }
 
-  if (patch_svc(memory.code, app->elf.load_size) != 0) {
+  uint32_t start = 0;
+  uint32_t code_length = app->elf.load_size;
+  if (app->elf.code_start && app->elf.code_end) {
+    start = app->elf.code_start - 0xc0de0000;
+    code_length = app->elf.code_end - app->elf.code_start;
+  }
```
Comment (fbeutin-ledger) [inline src/launcher.c:178]: Is there a way to factorize this duplicated code?

File: src/launcher.c (hunk 7 above)
Comment (cbouvet-ledger) [reply src/launcher.c:321]: i agree :bow:

**Review by xchapron-ledger — APPROVED** [review]

File: src/launcher.c [hunk 9]
```diff
@@ -293,10 +315,28 @@ static void *load_app(char *name)
...
   }
 
+  if (patch_svc(code + start, code_length) != 0) {
+    warn("could not patch svcs in app code");
+    goto error;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_EXEC) != 0) {
+    warn("could not update mprotect in rx mode for app");
+    _exit(1);
```
Comment (fbeutin-ledger) [inline src/launcher.c:337]: Why an `exit` here, and a `goto error` above ?

Comment (cbouvet-ledger) [reply src/launcher.c:337]: fixed in a later commit sorry

**Pull request merged**